src/edu/csus/ecs/pc2/ui/team/CreateTeamsTSV.java

```diff
@@ -0,0 +1,268 @@
+        System.exit(FAILURE_EXIT_CODE);
+
+    }
+
+    private static void usage() {
+        String[] usage = { //
+                "", //
+                "Usage CreateTeamsTSV [--help] [-f] [--of outputfilename] [--if teams_file.json]", //
+                "", //
+                "Purpose: read teams.json file create teams.tsv and groups.tsv files.", //
+                "", //
+                "This will not overwrite output file, use -f to overwrite output file (and "+GROUPS_TSV_FILENAME+")", //
+                "", //
+                "options:", //
+                "--of file  - alternate output filename, default output filename is " + TEAMS_TSV_FILENAME, //
+                "--if file  - alternate input JSON filename, default  filename is " + TEAMS_JSON_FILENAME, //
+                     "-f   overwrite output file, by default will not overwrite output file.", //
+                "", //
+                "On success exit code will be " + SUCCESS_EXIT_CODE, //
+                "Any other exit code is an error.", //
+                "", //
+        };
+
+        for (String s : usage) {
+            System.out.println(s);
+        }
+    }
+
+    protected void writeTSVFile() throws Exception {
+
+        if (!allowOverWrite  && new File(outputFileName).exists()) {
+            fatalError("Output file exists (" + outputFileName + "), will not overwrite.  Use " + OUTPUT_FILE_KEY + " to write to a different file");
+        }
+
+        List<TeamAccount> accounts = loadAccounts(inputFileName);
+
+        String [] header = { "Teams", "1" };
+
+        String headerLine = String.join(TAB, header);
+
+        FileWriter writer = (new FileWriter(new File(outputFileName)));
+
+        writer.write(headerLine + NL);
+
+        for (TeamAccount teamAccount : accounts) {
+            // TODO write teams.tsv output file
+            //            System.out.println(" debug 22 " + teamAccount.toJSON());
+
+            String groupId = "1";
+            if (teamAccount.getGroup_ids().size() > 0) {
+                groupId = teamAccount.getGroup_ids().get(0);
+            }
+            
+            Integer value = groupMap.get(groupId);
+            if (value == null) {
+                value = 0;
+            }
+            groupMap.put(groupId, value.intValue()+1);
+            
+            /**
+                     1  Team Number     22  integer
+2   External ID     24314   integer
+3   Group ID    4   integer
+4   Team name   Hoos    string
+5   Institution name    University of Virginia  string
+6   Institution short name  U Virginia  string
+7   Country Code    USA     string ISO 3166-1 alpha-3 
+             */
+            String [] fields = { //
+                    ""+teamAccount.getId(),
+                    teamAccount.getIcpc_id(), //
+                    groupId, //
+                    teamAccount.getName(), //
+                    teamAccount.getDisplay_name(), //
+                    "",
+                    "XXX", //
+            };
```

Comment:
I don't understand the reason for all the `//` comments at the end of each of the above lines.  They make no difference syntactically or semantically; they just make it harder to read the code.  And the fact that they are _present_ on some lines but _absent_ on others leaves the reader with further confusion...

Reply:
The // is a way to keep data on separate lines when a source formatter formats code.
A source formatter will typically join the values on a single line which can
make it a bit more difficult to quickly identify the fields.

Reply:
Aha, I see.  That doesn't explain why some lines end with // and others don't, but ok...